transport/src/main/java/io/netty/channel/SingleThreadEventLoop.java

```diff
@@ -71,11 +86,69 @@ public ChannelFuture register(final Channel channel, final ChannelPromise promis
         return promise;
     }
 
+    /**
+     * Adds a task to be run once at the end of next (or current) {@code eventloop} iteration.
+     *
+     * @param task to be added.
+     */
+    final void onEventLoopIteration(Runnable task) {
+        ObjectUtil.checkNotNull(task, "task");
+        if (isShutdown() || isShuttingDown() || isTerminated()) {
+            reject();
+        }
+
+        if (!tailTasks.offer(task)) {
+            throw new RejectedExecutionException("Task queue for onEventLoopIteration tasks is full.");
+        }
+
+        if (wakesUpForTask(task)) {
+            wakeup(inEventLoop());
+        }
+    }
+
+    /**
+     * Removes a task that was added previously via {@link #onEventLoopIteration(Runnable)}.
+     *
+     * @param task to be removed.
+     *
+     * @return {@code true} if the task was removed as a result of this call.
+     */
+    final boolean removeOnEventLoopIterationTask(Runnable task) {
+        return tailTasks.remove(ObjectUtil.checkNotNull(task, "task"));
+    }
+
     @Override
     protected boolean wakesUpForTask(Runnable task) {
         return !(task instanceof NonWakeupRunnable);
     }
 
+    @Override
+    protected boolean runAllTasks(long timeoutNanos) {
+        assert inEventLoop();
+        runTimeout = timeoutNanos;
+        if (lastRunTimedOut) {
+            lastRunTimedOut = false;
+            runAllTasksFrom(tailTasks, timeoutNanos, runTimeoutUpdater);
+        }
+
+        return super.runAllTasks(runTimeout);
+    }
+
+    @Override
+    protected void afterRunningAllTasks(boolean timedOut) {
+        if (timedOut) {
+            lastRunTimedOut = timedOut;
+            return;
+        }
+
+        runAllTasksFrom(tailTasks);
+    }
+
+    @Override
+    protected boolean hasTasks() {
```

Comment:
`SingleThreadEventExecutor` does not have it as final so I didn't make this final (It will break backward compatibility).